crates/move-prover-boogie-backend/src/boogie_backend/bytecode_translator.rs

```diff
@@ -3874,7 +3874,31 @@ impl<'env> FunctionTranslator<'env> {
                             true_expr_str,
                             false_expr_str
                         );
-                    }
+                    },
+                    Quantifier(qt, qid, inst) => {
+                        let fun_env = self.parent.env.get_function(*qid);
+                        let fun_name = boogie_function_name(&fun_env, inst, FunctionTranslationStyle::Default);
+
+                        let param_types = fun_env.get_parameter_types();
+                        let loc_type = param_types[0].skip_reference().instantiate(self.type_inst);
```

Comment:
the order should be:
* instantiate `inst` with `self.type_inst`
* use instantiated instance for `fun_name` (similar with function call)
* instantiate parameter type with instantiated instance